subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1534,6 +1525,34 @@ private void assertIsResolvable() {
         }
     }
 
+    private void assertIsDeclarable() {
+        if (!canBeDeclared) {
+            throw new IllegalStateException("Declaring dependencies for configuration '" + name + "' is not allowed as it is defined as 'canBeDeclared=false'.\n");
+        }
+    }
+
+    /**
+     * Ensure that the combination of consumable, resolvable and declarable flags is sensible.
+     * This method should be called only after all mutations are known to be complete.
+     */
+    private void assertIsProperConfiguration() {
```

Comment:
We only call this when we try to resolve a configuration. Is there another place we could call this, for example when someone tries to consume this configuration? Perhaps when we generate its metadata, wherever that is?

Reply:
Good idea.  I think if we check when we finalize the conf against further changes this will work.  I'll give it a try.